### PR TITLE
Automated / semi-automated python3 upgrades

### PR DESCRIPTION
**Pull request by asottile**
I split this into separate commits to make it easier to review, rebase, and replay.

### CODE REVIEW

File: src/_pytest/assertion/rewrite.py [hunk 1]
```diff
@@ -330,16 +328,16 @@ def _write_pyc(state, co, source_stat, pyc):
...
-        state.trace("error writing pyc file at %s: errno=%s" % (pyc, e.errno))
+        state.trace("error writing pyc file at {}: errno={}".format(pyc, e.errno))
         # we ignore any failure to write the cache file
         # there are many reasons, permission-denied, __pycache__ being a
         # file etc.
         return False
     return True
 
 
-RN = "\r\n".encode("utf-8")
```
Comment (nicoddemus) [inline src/_pytest/assertion/rewrite.py]: Not sure this is correct... doesn't `encode` use the default system locale?

**Comment by blueyed** [comment]
Why is "coding" removed again?

**Comment by asottile** [comment]
> 
> 
> Why is "coding" removed again? ([5e2e6b9](https://github.com/pytest-dev/pytest/commit/5e2e6b9288adeb7d7088d75962a4b8d96fcbf931))

https://www.python.org/dev/peps/pep-3120/

File: setup.py [hunk 2]
```diff
@@ -1,11 +1,9 @@
-# -*- coding: utf-8 -*-
 from setuptools import setup
 
 # TODO: if py gets upgrade to >=1.6,
 #       remove _width_of_current_line in terminal.py
 INSTALL_REQUIRES = [
     "py>=1.5.0",
-    "six>=1.10.0",
```
Comment (nicoddemus) [inline setup.py]: Thanks `six` for your service. 🙇

File: .pre-commit-config.yaml [hunk 3]
```diff
@@ -7,17 +7,18 @@ repos:
         args: [--safe, --quiet]
         language_version: python3
```
Comment (nicoddemus) [inline .pre-commit-config.yaml]: Didn't we have a pre-commit hook for the `encoding` comments? 🤔

Comment (asottile) [reply .pre-commit-config.yaml]: yep, and now it is enforcing that they are removed `fix-encoding-pragma` w/ `args: [--remove]`

File: src/_pytest/assertion/rewrite.py (hunk 1 above)
Comment (asottile) [reply src/_pytest/assertion/rewrite.py]: in python3+ it uses UTF-8 if no encoding is specified: https://github.com/asottile/pyupgrade/issues/138

File: setup.py (hunk 2 above)
Comment (asottile) [reply setup.py]: o7

/me pours one 🍺 out

File: testing/code/test_source.py [hunk 4]
```diff
@@ -113,7 +113,7 @@ def test_source_strip_multiline():
 def test_syntaxerror_rerepresentation():
     ex = pytest.raises(SyntaxError, _pytest._code.compile, "xyz xyz")
     assert ex.value.lineno == 1
-    assert ex.value.offset == 7
```
Comment (asottile) [inline testing/code/test_source.py]: damnit lol

```diff
commit 15669e81310e80018a86ed96678992e575f2f537 (HEAD -> pyupgrade)
Author: Anthony Sottile <asottile@umich.edu>
Date:   Sun Jun 2 18:02:14 2019 -0700

    Allow multiple positions for the SyntaxError in pypy3.6

diff --git a/testing/code/test_source.py b/testing/code/test_source.py
index 72bc628ab..a12a102a0 100644
--- a/testing/code/test_source.py
+++ b/testing/code/test_source.py
@@ -113,7 +113,7 @@ def test_source_strip_multiline():
 def test_syntaxerror_rerepresentation():
     ex = pytest.raises(SyntaxError, _pytest._code.compile, "xyz xyz")
     assert ex.value.lineno == 1
-    assert ex.value.offset == 7
+    assert ex.value.offset in {5, 7}  # cpython: 7, pypy3.6 7.1.1: 5
     assert ex.value.text.strip(), "x x"
 
 
```

Comment (asottile) [reply testing/code/test_source.py]: actually, I'm going to push mine since it has more context 🤷‍♂

**Comment by nicoddemus** [comment]
I don't understand why the coverage is reporting uncovered lines... for example, it says here:

https://codecov.io/gh/pytest-dev/pytest/compare/28ac469eaa7206a0bf1439b31ed187beb3598522...15669e81310e80018a86ed96678992e575f2f537/changes

That `test_tmpdir_fixture` was not executed at all, which doesn't really make sense... 🤔

**Comment by asottile** [comment]
I'm always suspicious of any data that `codecov` produces, in this case that test definitely ran: https://travis-ci.org/pytest-dev/pytest/jobs/540538554 🤔

**Comment by nicoddemus** [comment]
Hmmm indeed, but I noticed that `PYTEST_COVERAGE=1` was not set for that build. I will enable it for pure `py37` runs, let's see.

**Comment by nicoddemus** [comment]
Rebased on `master` (hope you don't mind @asottile) and fixed linting.

**Comment by asottile** [comment]
no problem, though I would have replayed the commits myself so that ended up in the right one :man_shrugging:

**Comment by nicoddemus** [comment]
OK I think we are good. Can you go ahead and merge @asottile if you are OK with the PR?

File: .travis.yml [hunk 5]
```diff
@@ -34,7 +34,7 @@ jobs:
         - test $(python -c 'import sys; print("%d%d" % sys.version_info[0:2])') = 37
 
     # Full run of latest supported version, without xdist.
-    - env: TOXENV=py37
+    - env: TOXENV=py37  PYTEST_COVERAGE=1
```
Comment (blueyed) [inline .travis.yml]: We do not need this, do we?
We have py37-xdist with coverage already (at least).

Comment (blueyed) [reply .travis.yml]: I'd guess it was the rebase that fixed it.

We can revert this and this - after things stabilized a bit.. :)

**Pull request merged**